src/main/scala/edu/ie3/simona/agent/participant/evcs/EvcsAgentFundamentals.scala

```diff
@@ -357,42 +353,188 @@ protected trait EvcsAgentFundamentals
         lastEvs
       )
 
-    val (power, evModelsCalculated) =
+    val (result, evModelsCalculated) =
       evcsModel.calculatePowerAndEvSoc(
-        currentTick,
+        tick,
         voltage,
         relevantData
       )
 
-    {
-      val departedEvs = calculateDepartedEvs(
-        evModelsCalculated,
-        evcsMovementsData.movements
-      )
-      if (departedEvs.nonEmpty) {
-        evServiceRef ! DepartedEvsResponse(
-          evcsModel.uuid,
-          departedEvs
-        )
+    val (departingEvs, stayingEvs) =
+      evModelsCalculated.partition { ev =>
+        // EV has been parked up until now and has now departed
+        requestedDepartingEvs.contains(ev.getUuid)
       }
-    }
 
-    val stayingEvs = calculateStayingEvs(
-      lastEvs,
-      evcsMovementsData.movements
-    )
     val updatedRelevantData = relevantData.copy(
       currentEvs = stayingEvs
     )
 
-    updateValueStoresInformListenersAndGoToIdleWithUpdatedBaseStateData(
-      scheduler,
+    if (departingEvs.nonEmpty) {
+      evServiceRef ! DepartingEvsResponse(
+        evcsModel.uuid,
+        departingEvs
+      )
+    }
+
+    /* Update the base state data */
+    updateValueStoresInformListeners(
       modelBaseStateData,
-      power,
+      tick,
+      result,
       updatedRelevantData
     )
   }
 
+  /** Handles a evcs movements message that contains information on arriving and
+    * departing vehicles. After applying the movements to the last known set of
+    * parked evs, calculates resulting charging power. Sends completion message
+    * to scheduler without scheduling new activations.
+    *
+    * @param tick
+    *   The current tick that has been triggered
+    * @param scheduler
+    *   The scheduler ref
+    * @param modelBaseStateData
+    *   The state data
+    * @param arrivingEvs
+    *   The movement data on arrivingEvs that has been received
+    * @return
+    *   [[Idle]] with updated result values
+    */
+  private def handleArrivingEvsAndGoIdle(
+      tick: Long,
+      scheduler: ActorRef,
+      modelBaseStateData: ParticipantModelBaseStateData[
+        ApparentPower,
+        EvcsRelevantData,
+        EvcsModel
+      ],
+      arrivingEvs: Seq[EvModel]
+  ): FSM.State[AgentState, ParticipantStateData[ApparentPower]] = {
+
+    val evcsModel = getEvcsModel(modelBaseStateData)
+
+    val (tickInterval, lastEvs) =
+      getTickIntervalAndLastEvs(tick, modelBaseStateData)
```

Comment:
s.o. (`currentEvs`)

Reply:
I didn't rename the variables, but left some clarifying commentary